internal/clients/kibana/alerting.go

```diff
@@ -0,0 +1,181 @@
+		}
+		req := client.CreateRule(ctxWithAuth, rule.SpaceID, "").KbnXsrf("true").CreateRuleRequest(reqModel)
+		ruleRes, res, err := req.Execute()
+		if err != nil && res == nil {
+			return nil, diag.FromErr(err)
+		}
+
+		defer res.Body.Close()
+		return ruleResponseToModel(rule.SpaceID, ruleRes), utils.CheckHttpError(res, "Unabled to create alerting rule")
+	}
+
+	reqModel := alerting.UpdateRuleRequest{
+		Actions:    actions,
+		Name:       rule.Name,
+		NotifyWhen: (*alerting.NotifyWhen)(&rule.NotifyWhen),
+		Params:     rule.Params,
+		Schedule: alerting.Schedule{
+			Interval: &rule.Schedule.Interval,
+		},
+		Tags:     rule.Tags,
+		Throttle: *alerting.NewNullableString(rule.Throttle),
+	}
+	req := client.UpdateRule(ctxWithAuth, rule.ID, rule.SpaceID).KbnXsrf("true").UpdateRuleRequest(reqModel)
+	ruleRes, res, err := req.Execute()
+	if err != nil && res == nil {
+		return nil, diag.FromErr(err)
+	}
+
+	defer res.Body.Close()
+	if diags := utils.CheckHttpError(res, "Unable to update alerting rule"); diags.HasError() {
+		return nil, diag.FromErr(err)
+	}
+
+	shouldBeEnabled := rule.Enabled != nil && *rule.Enabled
+	if shouldBeEnabled && !ruleRes.Enabled {
+		res, err := client.EnableRule(ctxWithAuth, rule.ID, rule.SpaceID).KbnXsrf("true").Execute()
+		if err != nil && res == nil {
+			return nil, diag.FromErr(err)
+		}
+
+		if diags := utils.CheckHttpError(res, "Unable to enable alerting rule"); diags.HasError() {
+			return nil, diag.FromErr(err)
+		}
+	}
+
+	if !shouldBeEnabled && ruleRes.Enabled {
+		res, err := client.DisableRule(ctxWithAuth, rule.ID, rule.SpaceID).KbnXsrf("true").Execute()
+		if err != nil && res == nil {
+			return nil, diag.FromErr(err)
+		}
+
+		if diags := utils.CheckHttpError(res, "Unable to disable alerting rule"); diags.HasError() {
+			return nil, diag.FromErr(err)
+		}
+	}
+
+	return ruleResponseToModel(rule.SpaceID, ruleRes), diag.Diagnostics{}
+}
+
+func GetAlertingRule(ctx context.Context, apiClient *clients.ApiClient, id, spaceID string) (*models.AlertingRule, diag.Diagnostics) {
+	client, err := apiClient.GetAlertingClient()
+	if err != nil {
+		return nil, diag.FromErr(err)
+	}
+
+	ctxWithAuth := apiClient.SetAlertingAuthContext(ctx)
+	req := client.GetRule(ctxWithAuth, id, spaceID)
+	ruleRes, res, err := req.Execute()
+	if err != nil && res == nil {
+		return nil, diag.FromErr(err)
+	}
+
+	defer res.Body.Close()
+	if res.StatusCode == http.StatusNotFound {
+		return nil, nil
+	}
+	return ruleResponseToModel(spaceID, ruleRes), utils.CheckHttpError(res, "Unabled to get alerting rule")
+}
+
+func DeleteAlertingRule(ctx context.Context, apiClient *clients.ApiClient, ruleId string, spaceId string) diag.Diagnostics {
+	client, err := apiClient.GetAlertingClient()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	ctxWithAuth := apiClient.SetAlertingAuthContext(ctx)
+	req := client.DeleteRule(ctxWithAuth, ruleId, spaceId).KbnXsrf("true")
+	res, err := req.Execute()
+	if err != nil && res == nil {
```

Comment:
The same question as for line #155

Reply:
https://github.com/elastic/terraform-provider-elasticstack/pull/292/files#r1145424005